### PR TITLE
Fixes #8693: Convert content view puppet module to scoped search.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb [hunk 1]
```diff
@@ -63,6 +56,14 @@ def destroy
       respond :resource => @puppet_module
     end
 
+    def index_relation
+      puppet_modules = ContentViewPuppetModule.where(:content_view_id => @view)
+      puppet_modules = puppet_modules.where(:name => params[:name]) if params[:name]
+      puppet_modules = puppet_modules.where(:uuid => params[:uuid]) if params[:uuid]
+      puppet_modules = puppet_modules.where(:author => params[:author]) if params[:author]
```
Comment (johnpmitsch) [inline app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb]: I see you have an author param in the index, but I don't see it declared in the model on scoped search. Did you want to do scoped search on author?

**Comment by johnpmitsch** [comment]
I was able to rebase locally so I could workaround the dependency error we talked about. When you rebase and push I can double check everything and then @jlsherrill can double check my review.

File: test/models/content_view_puppet_module_test.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      @puppet_module = ContentViewPuppetModule.find(katello_content_view_puppet_modules(:library_view_module_by_name).id)
+    end
+
+    def test_search_name
+      assert_equal @puppet_module, ContentViewPuppetModule.search_for("name = \"#{@puppet_module.name}\"").first
+    end
+
+    def test_search_content_view_name
+      assert_includes ContentViewPuppetModule.search_for("content_view_name = \"#{@library_view.name}\""), @puppet_module
+    end
```
Comment (johnpmitsch) [inline test/models/content_view_puppet_module_test.rb]: do you want to test for uuid search?

File: app/models/katello/content_view_puppet_module.rb [hunk 3]
```diff
@@ -13,6 +11,10 @@ class ContentViewPuppetModule < Katello::Model
 
     validates_with Validators::ContentViewPuppetModuleValidator
 
+    scoped_search :on => :name, :complete_value => true
+    scoped_search :on => :uuid, :complete_value => true
+    scoped_search :on => :name, :in => :content_view, :rename => :content_view_name
```
Comment (jlsherrill) [inline app/models/katello/content_view_puppet_module.rb]: might add author as well?

File: test/models/content_view_puppet_module_test.rb (hunk 2 above)
Comment (jlsherrill) [inline test/models/content_view_puppet_module_test.rb]: +1

**Comment by ehelms** [comment]
@johnpmitsch @jlsherrill Added the uuid and author tests (and author search).

File: test/models/content_view_puppet_module_test.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    def test_search_uuid
+      assert_includes ContentViewPuppetModule.search_for("content_view_name = \"#{@puppet_module.uuid}\""), @puppet_module
+    end
```
Comment (johnpmitsch) [inline test/models/content_view_puppet_module_test.rb]: just curious, Is the search key content_view_name when searching for uuid?

Comment (ehelms) [inline test/models/content_view_puppet_module_test.rb]: Good catch! Updated

**Comment by johnpmitsch** [comment]
Everything looks good to me. @jlsherrill I am having issues with my dev environment atm and can't pull up the UI to check search by author, could you look at it and then make the call to APJ or ACK?

**Comment by jlsherrill** [comment]
APJ

**Comment by ehelms** [comment]
[test]

**Pull request merged**